delfin/drivers/hitachi/vsp/rest_handler.py

```diff
@@ -0,0 +1,200 @@
+                    LOG.error('Login error,get access_session failed')
+            elif res.status_code == 503:
+                raise exception.InvalidResults(res.text)
+
+            return res
+
+        except Exception as e:
+            err_msg = "Get RestHandler.call failed: %s" % (six.text_type(e))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def get_resinfo_call(self, url, data=None):
+        result_json = None
+        res = self.call(url, data, 'GET')
+        if res.status_code == 200:
+            result_json = res.json()
+        return result_json
+
+    def login(self):
+        try:
+            self.get_device_id()
+            access_session = self.rest_auth_token
+            if self.san_address:
+                url = '%s%s/sessions' % \
+                      (RestHandler.COMM_URL,
+                       self.storage_device_id)
+                data = {}
+
+                with self.session_lock:
+                    if self.session is None:
+                        self.init_http_head()
+                    self.session.auth = \
+                        requests.auth.HTTPBasicAuth(
+                            self.rest_username,
+                            cryptor.decode(self.rest_password))
+                    res = self. \
+                        do_call(url, data, 'POST',
+                                calltimeout=consts.SOCKET_TIMEOUT)
+                    if res.status_code == 200:
+                        result = res.json()
+                        self.session_id = result.get('sessionId')
+                        access_session = 'Session %s' % result.get('token')
+                        self.rest_auth_token = access_session
+                        self.session.headers[
+                            RestHandler.AUTH_KEY] = access_session
+                    else:
+                        LOG.error("Login error. URL: %(url)s\n"
+                                  "Reason: %(reason)s.",
+                                  {"url": url, "reason": res.text})
+                        if 'invalid username or password' in res.text:
+                            raise exception.InvalidUsernameOrPassword()
+                        else:
+                            raise exception.BadResponse(res.text)
+            else:
+                LOG.error('Login Parameter error')
+
+            return access_session
+        except Exception as e:
+            LOG.error("Login error: %s", six.text_type(e))
+            raise e
+
+    def logout(self):
+        try:
+            url = RestHandler.LOGOUT_URL
+            if self.session_id is not None:
+                url = '%s%s/sessions/%s' % \
+                      (RestHandler.COMM_URL,
+                       self.storage_device_id,
+                       self.session_id)
+                if self.san_address:
+                    self.call(url, method='DELETE')
+                    self.rest_auth_token = None
```

Comment:
how about cleaning slef.device_id and serial number ?

Reply:
have changed that